### PR TITLE
fix integration test bug fix(fix #149)

### LINKED ISSUE #149: starknet integration test bug fix
1.The L2 private key generated through the private key and the wallet are different
2.The wallet signature method lacks the  change_pubkey_onchain_auth
3.add examples for approve/deposit/setAuthPubKeyHash
4.Expose pubkeyHash method

### PR DESCRIPTION
**Pull request by nick-zkp**

### CODE REVIEW

File: binding_tests/interface_test.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func TestSignWithdraw(t *testing.T) {
     fmt.Printf("zklink tx: %s\n", zklinkTx)
 
     // test signer
-    signer, err := sdk.NewSigner(s, sdk.L1TypeEth);
+    signer, err := sdk.NewSigner(s, sdk.L1TypeEth,nil,nil);
```
Comment (Fred0327) [inline binding_tests/interface_test.go:269]: add enum L1SignerType, and replace the `sdk.L1TypeEth, nil, nil`:

```
#[derive(Serialize, Deserialize)
#[serde(untagged)]
pub enum L1SignerType {
    Eth { l1: L1Type },
    Starknet { l1: L1Type, chain_id: String, address: String }
}
impl L1SignerType {
    pub fn eth_signer() -> Self;
    pub fn starknet_signer(chain_id, addr: String) -> Self;
}
```

File: signers/src/starknet_signer/typed_data/mod.rs [hunk 2]
```diff
@@ -118,7 +118,7 @@ impl TypedData {
         if let Ok(num) = BigUint::from_str_radix(s.trim_start_matches("0x"), 16) {
             format!("0x{}", num.to_str_radix(16))
         } else {
-            format!("0x{}", hex::encode(s))
+            format!("0x{}", hex::encode(s.replace('\n', "")))
```
Comment (Fred0327) [inline signers/src/starknet_signer/typed_data/mod.rs:121]: why need to replace `\n`?

File: examples/Javascript/node-example/1_change_pubkey.js [hunk 3]
```diff
@@ -41,9 +41,37 @@ async function testEcdsaAuth() {
...
+        console.error(error);
+    }
+
+}
+
 async function main() {
-    console.log(global);
-    await testEcdsaAuth();
+    //await testEcdsaAuth();
+    await testOnchainAuth();
```
Comment (Fred0327) [inline examples/Javascript/node-example/1_change_pubkey.js:74]: uncomment testEcdsaAuth

File: signers/src/starknet_signer/typed_data/mod.rs (hunk 2 above)
Comment (nick-zkp) [reply signers/src/starknet_signer/typed_data/mod.rs:121]: because encodeData in js will ignored the '\n'

**Review by Fred0327 — APPROVED** [review]

**Pull request merged**